packages/contracts/src/wnft/LibToken.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+
+import { ERC721System } from "@latticexyz/world-modules/src/modules/erc721-puppet/ERC721System.sol";
+import { IERC721Metadata } from "@latticexyz/world-modules/src/modules/erc721-puppet/IERC721Metadata.sol";
+import { WNFTOwnership } from "../codegen/index.sol";
+
+struct Token {
+  string name;
+  string symbol;
+}
+
+library LibToken {
+  error LibToken_MustBeTokenOwner();
+
+  function tokenOptions(address account) internal view returns (Token memory token) {
+    IERC721Metadata erc721 = _takeERC721(account);
+
+    string memory name = erc721.name();
+    string memory symbol = erc721.symbol();
+
+    token = Token({ name: name, symbol: symbol });
+  }
+
+  function tokenURI(uint256 tokenId, address account) internal view returns (string memory) {
+    IERC721Metadata erc721 = _takeERC721(account);
```

Comment:
there should be no `account` arg, get the token address from some global config table